plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerLaunchConfigurationDelegate.java

```diff
@@ -102,28 +103,32 @@ public void launch(ILaunchConfiguration configuration, String mode, final ILaunc
     console.clearConsole();
     console.activate();
 
-    new ServerLaunchMonitor(configuration, launch, server).engage();
+    int port = server.getAttribute(ServerPortExtension.SERVER_ATTRIBUTE_PORT,
```

Comment:
`ServerPortExtension` is a UI class; `SERVER_ATTRIBUTE_PORT` on the other hand is a core concept and should be available elsewhere.  I think it belongs on `LocalAppEngineServerBehaviour`.